modules/kmm-symbolic-links-for-in-tree-dependencies.adoc

```diff
@@ -0,0 +1,57 @@
+// Module included in the following assemblies:
+//
+// * hardware_enablement/kmm-kernel-module-management.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="kmm-symbolic-links-for-in-tree-dependencies_{context}"]
+
+= Symbolic links for in-tree dependencies
+
+Some kernel modules depend on other kernel modules that are shipped with the node's operating system.
+To avoid copying those dependencies into the kmod image, KMM mounts `/usr/lib/modules` into both the build and the
+worker pod's filesystems.
+
+By creating a symlink from `/opt/usr/lib/modules/[kernel-version]/[symlink-name]` to
+`/usr/lib/modules/[kernel-version]`, `depmod` can use the in-tree kmods on the building node's filesystem to resolve
+dependencies.
+
+At runtime, the worker pod extracts the entire image, including the `[symlink-name]` symbolic link.
+That link points to `/usr/lib/modules/[kernel-version]` in the worker Pod, which is mounted from the node's filesystem.
+`modprobe` can then follow that link and load the in-tree dependencies as needed.
+
+In the following example, `host` is the symbolic link name under `/opt/usr/lib/modules/[kernel-version]`:
+
+[source,dockerfile]
+----
+ARG DTK_AUTO
+
+FROM ${DTK_AUTO} as builder
+
+#
+# Build steps
+#
+
+FROM ubi9/ubi
+
+ARG KERNEL_FULL_VERSION
+
+RUN dnf update && dnf install -y kmod
+
+COPY --from=builder /usr/src/kernel-module-management/ci/kmm-kmod/kmm_ci_a.ko /opt/lib/modules/${KERNEL_FULL_VERSION}/
+COPY --from=builder /usr/src/kernel-module-management/ci/kmm-kmod/kmm_ci_b.ko /opt/lib/modules/${KERNEL_FULL_VERSION}/
+
+# Create the symbolic link
+RUN ln -s /lib/modules/${KERNEL_FULL_VERSION} /opt/lib/modules/${KERNEL_FULL_VERSION}/host
+
+RUN depmod -b /opt ${KERNEL_FULL_VERSION}
+----
+
+[NOTE]
+====
+`depmod` will generate dependency files based on the kernel modules present on the node that runs the kmod
```

Comment:
Suggestion to remove "will" based on our guidelines to avoid future tense. https://redhat-documentation.github.io/supplementary-style-guide/#will

```suggestion
`depmod` generates dependency files based on the kernel modules present on the node that runs the kmod
```

Reply:
changed.